src/LogBackend.cxx

```diff
@@ -62,28 +71,56 @@ SetLogThreshold(LogLevel _threshold) noexcept
 	log_threshold = _threshold;
 }
 
+template <typename DurationType>
+inline constexpr const char *date_format_of = "";
+
+template<>
+inline constexpr const char *date_format_of<std::chrono::minutes> = "{:%b %d %H:%M} : ";
+template<>
+inline constexpr const char *date_format_of<std::chrono::seconds> = "{:%b %d %H:%M:%S} : ";
+template<>
+inline constexpr const char *date_format_of<std::chrono::milliseconds> = "{:%b %d %H:%M:%S} : ";
+
+template<typename DurationType>
+static const char *
+log_date(char buf[LOG_DATE_BUF_SIZE])
+{
+	using namespace std::chrono;
+
+	/*
+	 * NOTE: fmt of chrono::time_point with %S writes
+	 *       a floating point number of seconds if the
+	 *       duration resolution is less than second
+	 */
+	auto [p, n] = fmt::format_to_n(buf, LOG_DATE_BUF_SIZE, date_format_of<DurationType>,
+				       floor<DurationType>(system_clock::now()));
+	assert(n < LOG_DATE_BUF_SIZE);
+	*p = 0;
+	return buf;
+}
+
+static auto
```

Comment:
Why `auto`? Having `auto` here and `-> const char*` later is more complicated than just writing `const char *` here.